include/jwt-cpp/jwt.h

```diff
@@ -846,6 +895,164 @@ namespace jwt {
 			return pkey;
 		}
 
+		/**
+		* \brief create public key from modulus and exponent
+		*
+		* \param modulus	string containing base64 encoded modulus
+		* \param exponent	string containing base64 encoded exponent
+		* \param decode 				The function to decode the certs)
+		* \param ec			error_code for error_detection (gets cleared if no error occur
+		* \return public key in PEM format
+		*/
+		template<typename Decode>
+		std::string create_public_key_from_rsa_components(const std::string& modulus, const std::string& exponent,
+														  Decode decode, std::error_code& ec) {
+			auto decoded_modulus = decode(modulus);
```

Comment:
Would it make sense to pull out the decode template and put that burden on the caller ?
It's just a small nitpick, but it will reduce the binary size of apps that have lots of different decode functions.

Reply:
In my mind, we will get an external definition of public key in the jwks and jwk format and creating the public key is already a helper function. As such, I think using another helper (twice) to a helper function is a little cumbersome.

I am not concerned about compiled code size. These days it can be a concern in embedded systems that are tiny but nothing is tiny anymore

Reply:
I suspect if they are at the level of binary size being a concern they will have already implemented thier own highly optimized base64 decoder and if they call that one - they do not pay the price if the lambda.

This was originally a way to keep the API stable while also keeping the base.h header as optional 

I would definitely be will to change that for the next release if we want to simplify or separate some of the code to new headers